### ISSUE TITLE
Inconsistencies between rest api and C# api

### ISSUE THREAD

**Issue by andyfensham** [DAXGRID/typesense-dotnet#82]
When I create a collection for auto fields using Postman and the Rest API on v0.22.2,  like the following :

Post request to {{url}}/collections
```
{
  "name": "countries",  
  "fields": [
    {"name": ".*", "type": "auto" }
  ]
}
```

it works, but doing the same to the c# interface, does not work.  If I do
```
 var schema = new Schema(
         request.Name,
           new List<Field>
           {
                new Field(".*", FieldType.Auto,false)
           });
```

The response I get back is

```
{
    "responseStatus": {
        "errorCode": "TypesenseApiBadRequestException",
        "message": "{\"message\": \"Field `.*` must be an optional field.\"}",
   }
}

```

**Comment by runeanielsen** [comment]
Hi @andyfensham,

Optional is by default set to false. If you parse in `true` as the fourth parameter to the `Field` constructor, the field will be set as optional, and that should fix your issue.

Here is an example:

```c#
var schema = new Schema(
    "countries",
    new List<Field>
    {
        new Field(".*", FieldType.Auto, false, true)
    });
```

**Comment by runeanielsen** [comment]
Hi @andyfensham,

This has now been fixed in #85. I've made a new released `4.0.2` where it is fixed.

Thanks for opening the issue.

// Rune
